internal/ethapi/api.go

```diff
@@ -1453,15 +1453,23 @@ func newRPCPendingTransaction(tx *types.Transaction, current *types.Header, conf
 func newRPCTransactionFromBlockIndex(b *types.Block, index uint64, config *params.ChainConfig, db ethdb.Database) *RPCTransaction {
 	txs := b.Transactions()
 
-	borReceipt := rawdb.ReadBorReceipt(db, b.Hash(), b.NumberU64())
-	if borReceipt != nil {
-		tx, _, _, _ := rawdb.ReadBorTransaction(db, borReceipt.TxHash)
+	if index >= uint64(len(txs)+1) {
```

Comment:
Won't it suppress working with Bor transactions?

Reply:
Earlier `if index == uint64(len(txs))` check wasn't there and hence we added check on top before even making a call. I think with this check being added, we can remove this `if index >= uint64(len(txs)+1)` totally. But to answer your question, the check if against `len(txs)+1` so it keeps a buffer incase state-sync transactions are present.

Reply:
The check also prevents out of range index, which could be malicious, from invoking ReadBorReceipt or ReadBorTransaction.